receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -80,6 +80,9 @@ func (tsm *timeseriesMap) get(
 	metric *metricspb.Metric, values []*metricspb.LabelValue) *timeseriesinfo {
 	name := metric.GetMetricDescriptor().GetName()
 	sig := getTimeseriesSignature(name, values)
+
+	tsm.Lock()
```

Comment:
This was added as replacement to the one removed from AdjustMetrics since there was a chain until the tsm was actually used. @dinooliva let me know if there is a risk on this move.

Reply:
This will work but will add a lot of overhead to metrics adjustment, which will add latency to the critical path. I think leaving the lock where it was is better - it may add some latency to gc() but that shouldn't be an issue.

Reply:
If I see this right is because on the location that I proposed it will be acquired and released multiple times in a loop. Perhaps it can be moved to `MetricsAdapter.adjustMetricTimeSeries`since it may not need to acquire the lock (it is only needed if metric is not gauge). Anyway, I will follow your recommendation.